test/spec/module.test.ts

```diff
@@ -1,28 +1,70 @@
-import { Component } from '@angular/core';
+import { Component, APP_INITIALIZER } from '@angular/core';
 import { TestBed } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
-import { AuthSdkError, OktaAuth } from '@okta/okta-auth-js';
+import { HttpClient } from '@angular/common/http';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { OktaAuth, OktaAuthOptions } from '@okta/okta-auth-js';
 import { 
   OktaAuthModule, 
   OKTA_CONFIG, 
   OKTA_AUTH,
   OktaAuthStateService, 
-  OktaAuthGuard 
+  OktaAuthGuard,
+  OktaAuthConfigService
 } from '../../lib/src/okta-angular';
 
-jest.mock('../../lib/src/okta/packageInfo', () => ({
-  __esModule: true,
-  default: {
-    authJSMinSupportedVersion: '5.3.1',
-    version: '99.9.9',
-    name: '@okta/okta-angular',
-  }
-}));
-
 @Component({ template: '' })
 class MockComponent {}
 
-function setup(oktaAuth: OktaAuth) {
+// Simulate fetching OktaAuthOptions from backend with GET /config
+// In APP_INITIALIZER factory the config should be set with configService.setConfig()
+async function setupWithAppInitializer(oktaAuthOptions?: OktaAuthOptions) {
+  const configInitializer = (configService: OktaAuthConfigService, httpClient: HttpClient) => {
+    return () => {
+      return httpClient.get('/config')
+        .toPromise()
```

Comment:
Is there a reason to convert to a promise? `toPromise` is deprecated in v7 and intended to be removed in v8 I think.